x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/deployment/PyTorchResult.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.inference.deployment;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.ml.utils.MlParserUtils;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/*
+ * TODO This does not necessarily belong in core. Will have to reconsider
+ * once we figure the format we store inference results in client calls.
+*/
+public class PyTorchResult implements ToXContentObject, Writeable {
+
+    private static final ParseField REQUEST_ID = new ParseField("request_id");
+    private static final ParseField INFERENCE = new ParseField("inference");
+    private static final ParseField ERROR = new ParseField("error");
+
+    public static final ConstructingObjectParser<PyTorchResult, Void> PARSER = new ConstructingObjectParser<>("pytorch_result",
+        a -> new PyTorchResult((String) a[0], (double[][]) a[1], (String) a[2]));
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), REQUEST_ID);
+        PARSER.declareField(ConstructingObjectParser.optionalConstructorArg(),
+            (p, c) -> {
+                List<List<Double>> listOfListOfDoubles = MlParserUtils.parseArrayOfArrays(
+                    INFERENCE.getPreferredName(), XContentParser::doubleValue, p);
+                double[][] primitiveDoubles = new double[listOfListOfDoubles.size()][];
+                for (int i = 0; i < listOfListOfDoubles.size(); i++) {
+                    List<Double> row = listOfListOfDoubles.get(i);
+                    double[] primitiveRow = new double[row.size()];
```

Comment:
```suggestion
                    primitiveDoubles[i] = row.toArray(new double[]{});
```

rather than copying the elements. Or if the elements must be copied perhaps `System.arrayCopy(primitiveRow, row.toArray(new double[]{});`

Reply:
What about using the `row.stream().mapToDouble(d -> d).toArray();` way?